langchain/src/tools/base.ts

```diff
@@ -1,39 +1,39 @@
-import { CallbackManager, getCallbackManager } from "../callbacks/index.js";
-
-const getVerbosity = () => false;
+import { CallbackManager } from "../callbacks/base.js";
+import { BaseLangChain } from "../base_language/index.js";
 
 export interface ToolParams {
   verbose?: boolean;
   callbackManager?: CallbackManager;
 }
 
-export abstract class Tool {
-  verbose: boolean;
-
-  callbackManager: CallbackManager;
-
+export abstract class Tool extends BaseLangChain {
   constructor(verbose?: boolean, callbackManager?: CallbackManager) {
-    this.verbose = verbose ?? (callbackManager ? true : getVerbosity());
-    this.callbackManager = callbackManager ?? getCallbackManager();
+    super(verbose, callbackManager);
   }
 
-  protected abstract _call(arg: string): Promise<string>;
-
-  async call(arg: string, verbose?: boolean): Promise<string> {
-    const _verbose = verbose ?? this.verbose;
-    await this.callbackManager.handleToolStart(
-      { name: this.name },
-      arg,
-      _verbose
-    );
+  protected abstract _call(
+    arg: string,
+    callbackManager?: CallbackManager
+  ): Promise<string>;
+
+  async call(
+    arg: string,
+    _verbose?: boolean,
+    callbackManager?: CallbackManager
+  ): Promise<string> {
+    const localCallbackManager = this.configureCallbackManager(callbackManager);
+    await localCallbackManager?.handleToolStart({ name: this.name }, arg);
```

Comment:
Wouldn't it make more sense to return a bound callback manager copy from `handleToolStart` rather than have 5 lines of code here?